### PR TITLE
feat(IR-3436): make label prop of Checkbox component as string | Reac…

### PR DESCRIPTION
**Pull request by kattsushi**
…tNode

## Summary

Also updated the texts to match https://www.figma.com/design/a3qQCRxNWxP7K1titmso0e/Console-Design?node-id=5880-19887&t=8En1eQEUiLF2s3bi-0

## References
closes #_insert number here_

### CODE REVIEW

File: packages/ui/src/primitives/tailwind/Checkbox/index.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ import Label from '../Label'
 
 export interface CheckboxProps extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'value' | 'onChange'> {
   value: boolean
-  label?: string
+  label?: string | React.ReactNode
```
Comment (CITIZENDOT) [inline packages/ui/src/primitives/tailwind/Checkbox/index.tsx]: If label is a ReactNode, it should be rendered as it is at Line 63 (not inside the Label component).

**Review by CITIZENDOT — CHANGES_REQUESTED** [review]

File: packages/ui/src/primitives/tailwind/Checkbox/index.tsx (hunk 1 above)
Comment (aditya-mitra) [reply packages/ui/src/primitives/tailwind/Checkbox/index.tsx]: also type of label will be `label?: React.ReactNode` only since string is a sub type

**Review by aditya-mitra — CHANGES_REQUESTED** [review]
as per the [component parity document](https://tsu.atlassian.net/wiki/x/AQC5j), label will always be a string [like shown here](https://arc.net/l/quote/lqyxrken)

after checking on IR-3436, it seems like the [required design according to figma](https://www.figma.com/design/a3qQCRxNWxP7K1titmso0e/Console-Design?node-id=5880-19887&t=n1SR7ip06vY9bboJ-0) is possible on the current checkbox

**Comment by hanzlamateen** [comment]
> as per the [component parity document](https://tsu.atlassian.net/wiki/x/AQC5j), label will always be a string [like shown here](https://arc.net/l/quote/lqyxrken)
> 
> after checking on IR-3436, it seems like the [required design according to figma](https://www.figma.com/design/a3qQCRxNWxP7K1titmso0e/Console-Design?node-id=5880-19887&t=n1SR7ip06vY9bboJ-0) is possible on the current checkbox

@aditya-mitra we do need to add a link to term of service as done for the profile menu, Otherwise how can a user find/read terms of service. So we will need to make it ReactNode

image

File: packages/ui/src/primitives/tailwind/Checkbox/index.tsx (hunk 1 above)
Comment (hanzlamateen) [reply packages/ui/src/primitives/tailwind/Checkbox/index.tsx]: Yes, we should use `label?: React.ReactNode` instead of `label?: string | React.ReactNode`

**Review by hanzlamateen — APPROVED** [review]

**Pull request merged**